core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -213,6 +217,14 @@ object ConsoleConsumer extends Logging {
       .withRequiredArg
       .describedAs("blacklist")
       .ofType(classOf[String])
+    val partitionIdOpt = parser.accepts("partition", "The partition to consume from.")
+      .withRequiredArg
+      .describedAs("partition")
+      .ofType(classOf[java.lang.Integer])
+    val offsetOpt = parser.accepts("offset", "The offset id to consume from, default to -2 which means from beginning; while value -1 means from end")
```

Comment:
1. Yes, will add.
2. I'll update and change it to a String input with proper processing.